packages/next/build/index.ts

```diff
@@ -523,6 +527,40 @@ export default async function build(dir: string, conf = null): Promise<void> {
   )
   staticCheckWorkers.end()
 
+  if (serverPropsPages.size > 0) {
+    // We update the routes manifest after the build with the
+    // serverProps routes since we can't determine this until after build
+    routesManifest.serverPropsRoutes = {}
+
+    for (const page of serverPropsPages) {
+      const cleanDataRoute = path.posix.join(
+        '/_next/data',
+        buildId.replace(/[|\\{}()[\]^$+*?.-]/g, '\\$&'),
```

Comment:
We've cloned this regex in so many places in the project, maybe it's time to just import it from https://github.com/sindresorhus/escape-string-regexp?

Reply:
Opened PR here https://github.com/zeit/next.js/pull/10257